### PR TITLE
RIA-3116: Changes to support final bundling integration with advanced…

### PR DESCRIPTION
**Pull request by jkumar-moj**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3116

### Change description ###
Changes to support final bundling integration with advanced bundling 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AdvancedBundlingCallbackHandler.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            .stream()
+            .map(IdValue::getValue)
+            .collect(Collectors.toList());
+
+        if (caseBundles.size() != 1) {
+            throw new IllegalStateException("case bundles size is not 1 and is : " + caseBundles.size());
+        }
+
+        //stictchStatusflags -  NEW, IN_PROGRESS, DONE, FAILED
+        final String stitchStatus = caseBundles.get(0).getStitchStatus().orElse("");
```
Comment (makowalski) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AdvancedBundlingCallbackHandler.java:83]: same as it was in notifications-api, we need to figure it out later for more than one bundle usage

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SendNotificationHandlerTest.java [hunk 2]
```diff
@@ -170,7 +170,7 @@ public void it_can_handle_callback() {
                         Event.REQUEST_RESPONSE_REVIEW,
                         Event.SUBMIT_CASE,
                         Event.SEND_DECISION_AND_REASONS,
-                        Event.GENERATE_HEARING_BUNDLE,
+                        //Event.GENERATE_HEARING_BUNDLE,
```
Comment (makowalski) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SendNotificationHandlerTest.java:173]: we can remove if not needed

Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SendNotificationHandlerTest.java]: Has been removed

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AdvancedBundlingCallbackHandler.java (hunk 1 above)
Comment (jkumar-moj) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AdvancedBundlingCallbackHandler.java:83]: Agree Marcin. That would also depend on the UX design whether we plan to put alll bundles together or in separate collections. We can handle it later I think.

**Comment by makowalski** [comment]
changes have been merged together with RIA-2923

**Pull request closed without merging**